src/sites/dMCDA.jl

```diff
@@ -818,7 +818,7 @@ function run_site_selection(domain::Domain, criteria::DataFrame, sum_cover::Abst
             sum_cover[cover_ind, :, :],
             area_to_seed
         )
-        ranks_store[cover_ind, 1:size(ranks_temp, 1), :] = ranks_temp
+        ranks_store[cover_ind, depth_priority, :] = ranks_temp
```

Comment:
Is there a potential for mismatch here as `depth_priority` relies on index position?

Also note that ReefMod doesn't use depth, so when developing the ReefMod interface I simply set depth_priority to `1:nrow(site_data)` - need a better way of handling that case

Reply:
Yeah there is, I'd have to change things in site_selection though because I think the set is depth filtered first before being ranked, so the set that comes out has the size of the filtered set rather than the full set. I can look into it.

Reply:
On reviewing, I think the main way a mismatch would occur is if the site_data had duplicate values in site_ids, which it shouldn't. I've simplified the definition of depth_priority a bit anyway (using findall instead of the vector of ids), so it should correspond directly to the site data.

Reply:
I may be in need of a second coffee but I'm not entirely sure how using `findall` helps here. I don't think it hurts, but I can't see how it is advantageous.

Reply:
It's just neater, no difference

Reply:
But I'm not sure how else to index here, as `depth_priority` is what is used inside `site_selection`. I suppose we could use `ranks_temp[:,1]` but it should be the same

Reply:
I see what you mean re neatness.

Here's my understanding of how this is structured:

`depth_priority` indexes the full site list by position (the variable name could be clearer, but ignore that for the moment).

Inside `site_selection()`, it is used to subset the full site list again.

This is then used to index `ranks_store`.

This will work (has been working?) so long as nothing gets accidentally reordered, the index positions correctly map back to the full index, and there are no bugs when slicing arrays. But there are several places where arrays are indexed/subsetted etc.

Given `ranks_store` is a NamedDimsArray, you could start by using the actual location names/site ids as the index keys. This will still work with positional indexing. Then, you could change the positional indexing to use location names/site ids. This will protect from mismatches in a more robust way than simply relying on matching array sizes and positions.

Reply:
Let me know if what I just pushed is better

Reply:
Looks like a good start. Does the example run without issue now?

FYI, I was puzzled as to why GitHub wasn't auto-linking the issue you mentioned and it seems "addresses" is not a closing keyword: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Reply:
Sorry I forgot that was the main issue here ... I looked into it again and I think maybe we're working with different definitions here - the way I've set it up is so that coral cover has dimensions (:scenarios,:sites,:species), to allow the possibility of having different initial coral covers for different scenarios. In the example you sent coral_cover has dimensions (:sites,:species)... I can make it a single coral cover layer if you think thats too complicated.

Also, apologies for not having a key word, I didn't realise you had to have that as well as the issue hash.

Reply:
> apologies for not having a key word, I didn't realise you had to have that as well as the issue hash.

Meanwhile, I thought "addresses" was a keyword

> to allow the possibility of having different initial coral covers for different scenarios. In the example you sent coral_cover has dimensions (:sites,:species)

Sure but the example above is based on your example (in the examples folder, which should eventually be moved into the docs)

Reply:
Right, I must have updated the test but not the example (weird, I thought I had). Should I keep it the way it is and have multiple initial coral covers? Or do you reckon one is best?

Reply:
You could support both, but to keep things simple for now, let's have just one.

ADRIA uses the mean connectivity values (with a view to move towards a stochastic representation).
ReefMod has multiple connectivity years, but I'm just taking the mean of those for now.

Reply:
Ok cool, also what was the example script called that you are working from?

Reply:
`examples/site_selection.jl`

Reply:
Did you change the example? it looks different from the one in Main and site-selection-sampling-error
https://github.com/open-AIMS/ADRIA.jl/blob/0808ba70cb4f1e3192f34408dfcfc62a407a5245/examples/site_selection.jl#L16

Reply:
Anyway, I'll change it to what you have

Reply:
Yes, I am using a modified version of the example to work on the ReefMod interface.

In the example you linked, you're creating a matrix of $N \cdot S$, where $N$ is the number of scenarios and $S$ is the number of locations.

This line: `sum_cover = repeat(sum(dom.init_coral_cover, dims=2)', size(scens, 1))`

Does the same, using the initial coral cover data loaded for the domain

Reply:
When I infiltrated it seemed like it was scenarios*species (8 by 36), but scenarios was labelled with sites. What I've done is keep it so we can still have different coral cover scenarios, but it sums over species instead.. There was also an issue with it trying to index species, I think previously the summation was done in site_selection() but we took it out (I'm not sure why it didn't throw an error in tests, but I'll add a test for mcda_vars() to check the components are the correct size

Reply:
Ok tests are passing and example works now

Reply:
I think this is ready for final review when you have time @ConnectedSystems . Tests passing on my end.